include/deal.II/matrix_free/cuda_matrix_free.templates.h

```diff
@@ -137,7 +143,10 @@ namespace CUDAWrappers
 
       template <typename CellFilter>
       void
-      get_cell_data(const CellFilter &cell, const unsigned int cell_id);
+      get_cell_data(
+        const CellFilter &                                 cell,
+        const unsigned int                                 cell_id,
+        std::shared_ptr<const Utilities::MPI::Partitioner> partitioner);
```

Comment:
Let us just pass by `const &` as usual.

Reply:
But then I also need to pass a flag to know if the partitioner was initialized or not. This way I can just check if the pointer is `nullptr`.

Reply:
Oh, I meant passing a `const std::shared_ptr<const Utilities::MPI::Partitioner>&` and not `const Utilities::MPI::Partitioner &`.